sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -134,4 +155,91 @@ export class StudentController extends BaseController {
   ): Promise<void> {
     await this.studentService.synchronizeFromUserInfo(userToken);
   }
+
+  /**
+   * Allow files uploads to a particular student.
+   * @param userToken authentication token.
+   * @param file file content.
+   * @param uniqueFileName unique file name (name+guid).
+   * @param groupName friendly name to group files. Currently using
+   * the value from 'Directory' property from form.IO file component.
+   * @returns created file information.
+   */
+  @Post("files")
+  @UseInterceptors(
+    FileInterceptor("file", {
+      limits: uploadLimits(MAX_UPLOAD_FILES, MAX_UPLOAD_PARTS),
+      fileFilter: defaultFileFilter,
+    }),
+  )
+  async uploadFile(
+    @UserToken() userToken: IUserToken,
+    @UploadedFile() file: Express.Multer.File,
+    @Body("uniqueFileName") uniqueFileName: string,
+    @Body("group") groupName: string,
+  ): Promise<FileCreateDto> {
+    const student = await this.studentService.getStudentByUserName(
```

Comment:
do we need to check student exceptions here?

Reply:
Yes, I will add a check.

Reply:
Checks added.